gvsbuild/projects/gstreamer.py

```diff
@@ -306,3 +306,47 @@ def __init__(self):
     def build(self):
         Meson.build(self)
         self.install(r".\COPYING share\doc\gst-rtsp-server")
+
+@project_add
+class GstPluginGtk4(Tarball, Project):
+    def __init__(self):
+        Project.__init__(
+            self,
+            "gst-plugin-gtk4",
+            version="0.14.3",
+            repository="https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs",
+            archive_url="https://crates.io/api/v1/crates/gst-plugin-gtk4/{version}/download",
+            archive_filename="gst-plugin-gtk4-{version}.tar.gz",
+            hash="defe41c5b45f2f1617253274abcaa497b53624fd6e23c1d89fe5e0c8cfbe3fd2",
+            dependencies=["cargo", "gst-plugins-base", "gtk4"],
+        )
+
+    def build(self):
+        self.builder.exec_cargo("install cargo-c")
+        print("gst-plugin-gtk4: installed cargo-c")
+        print("gst-plugin-gtk4: TODO: Translate this into gvsbuild: cargo cinstall -p gst-plugin-gtk4 --prefix 'C:\gtk-build\build\gtk\x64\release' --libdir 'C:\gtk-build\build\gtk\x64\release\lib'")
+        # TODO: then test via gst-inspect-1.0.exe gtk4paintablesink
+
+
+@project_add
+class GstPluginGtk4Meson(Tarball, Meson):
+    def __init__(self):
+        Project.__init__(
+            self,
+            "gst-plugin-gtk4-meson",
+            version="0.14.3",
+            repository="https://gitlab.freedesktop.org/gstreamer/gst-plugins-rst",
+            archive_url="https://gitlab.freedesktop.org/gstreamer/gst-plugins-rs/-/archive/{version}/gst-plugins-rs-{version}.tar.gz?ref_type=tags",
+            archive_filename="gst-plugins-rs-{version}.tar.gz",
+            hash="0609b2459ec29f4678edd98e69b6b0a473ef4a303d60645260245dbc23f75167",
+            dependencies=["meson", "cargo", "gst-plugins-base", "gtk4"],
```

Comment:
gstreamer-webrtc should be in gst-plugins-bad, so you can try to add that as a dependency.